content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx

```diff
@@ -225,3 +225,77 @@ const user = await prisma.user.findUnique({
   },
 })
 ```
+
+#### Filter on composite types
+
+<Admonition type="warning">
+
+Composite types are only available in MongoDB.
+
+</Admonition>
+
+In Prisma versions 4.3.0 and later, you can search on a multi-field constraint that contains composite types. For example, the following schema has a `Product` model, an embedded `Photo` composite type, and a multi-field unique constraint on the product `name` and photo `url`:
+
+```prisma file=schema.prisma
+model Product {
+  id          String  @id @default(auto()) @map("_id") @db.ObjectId
+  name        String  @unique
+  description String
+  photos      Photo[]
+
+  @@unique([name, photos.url]) // multi-field unique constraint
+}
+
+type Photo {
+  caption  String
+  url      String
+}
+```
+
+In this case , the default API to select on the constraint uses a generated combination of the model field name, and the type name and field. For example, the following query searches on `name_photo_url` to get the `Product` record with a given `name`, and containing a `Photo` composite type with a given `url`:
+
+```ts
+const product = await prisma.product.findUnique({
+  where: {
+    name_photo_url: {
```

Comment:
Its worth nothing that users must use both the `name` and `photo` predicates when querying the index. For example: 

We don't allow this: 

```ts
const product = await prisma.product.findUnique({
  where: {
    name_photo_url: {
      name: 'Forest Runners'
    },
  },
})
```

Nor do we allow this: 

```ts
const product = await prisma.product.findUnique({
  where: {
    name_photo_url: {
      photo: {
        url: '1.jpg',
      },
    },
  },
})
```

Reply:
Thanks. I've now added an example to cover this limitation.